src/charm.py

```diff
@@ -63,6 +64,18 @@ class ProxyMode(StrEnum):
     INVALID = "invalid"
 
 
+def _validate_port(port: int) -> bool:
```

Comment:
Out of scope but it would be nice if libraries validate data before adding to the relation so provider/requirer don't need to do this.

Reply:
The legacy relation doesn't  have any data validation and we're not trying to bring it up to date, it's just here to support drop-in replacement. So we decided to try and at least do some data validation for the legacy interface but anyone using this charm should use either the `ingress` or `haproxy-route` interface rather than this legacy one.